shard_consumer.go

```diff
@@ -63,7 +64,10 @@ func getRecords(k kinesisiface.KinesisAPI, iterator string) (records []*kinesis.
 		return nil, "", 0, err
 	}
 
-	records = output.Records
+	records, err = deaggregator.DeaggregateRecords(output.Records)
+	if err != nil {
```

Comment:
Not sure about swallowing this error. What are the possible errors from deaggregator?

Reply:
@GarethLewin deaggregator throws error in case if it fails to unmarshal the de aggregated record.
`err := proto.Unmarshal(messageData, aggRecord)

				if err != nil {
					return nil, err
				}`
Should we not return the actual payload without deaggregation in case of such error, as it can be possible the records have been aggregated using some custom logic not via amazon's aggregation format, so for those scenarios we should return the records as pushed. It should be up to user to deaggregate them in that case.